### PR TITLE
Benchmarks: Add Feature - Add reduce function support for output summary.

### PR DESCRIPTION
**Pull request by guoshzhao**
**Description**
Add reduce function support for output summary.

**Major Revision**
- Add reducer class to maintain all reduce functions.
- Save reduce type of each metric into `BenchmarkResult`
- Fix UT.

### CODE REVIEW

File: superbench/benchmarks/reducer.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    """The Enum class representing different reducer."""
+    AVG = 'avg'
+    MAX = 'max'
+    MIN = 'min'
+    SUM = 'sum'
+
+
+class Reducer:
+    """Reducer class to maintain all reduce functions."""
+    functions: Dict[ReduceType, Callable] = dict()
```
Comment (abuccts) [inline superbench/benchmarks/reducer.py:22]: why not write the content here directly?

File: superbench/benchmarks/micro_benchmarks/micro_base.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def _process_numeric_result(self, metric, result):
             return False
 
         self._result.add_raw_data(metric, result)
-        self._result.add_result(metric, sum(result) / len(result))
+        self._result.add_result(metric, sum(result) / len(result), reduce_type)
```
Comment (abuccts) [inline superbench/benchmarks/micro_benchmarks/micro_base.py:71]: built-in `sum` has [approximation issue](https://docs.python.org/3/tutorial/floatingpoint.html), e.g.,
```
>>> sum([0.1, 0.1, 0.1])
0.30000000000000004
```
```suggestion
        self._result.add_result(metric, statistics.mean(result), reduce_type)
```

File: superbench/benchmarks/reducer.py (hunk 1 above)
Comment (guoshzhao) [reply superbench/benchmarks/reducer.py:22]: Want to add reduce functions in a unified way by calling `add_reduce_func()`:

```
@Reducer.add_reduce_func(ReduceType.PERCENT99)
def percent_99(items):
    xxx
```

File: superbench/benchmarks/micro_benchmarks/micro_base.py (hunk 2 above)
Comment (guoshzhao) [reply superbench/benchmarks/micro_benchmarks/micro_base.py:71]: I will correct it in a separate PR. Thanks!

**Review by cp5555 — APPROVED** [review]

**Review by abuccts — APPROVED** [review]

**Pull request merged**